pkg/docker/containers-view.jsx

```diff
@@ -371,60 +417,98 @@ var ImageList = React.createClass({
         this.setState({ pulling: this.props.client.pulling });
     },
 
+    vulnerableInfoChanged: function (event, infos) {
+        this.setState({ vulnerableInfos: infos });
+    },
+
     componentDidMount: function () {
         $(this.props.client).on('image.containers', this.imagesChanged);
         $(this.props.client).on('pulling.containers', this.pullingChanged);
+
+        atomic.addEventListener('vulnerableInfoChanged', this.vulnerableInfoChanged);
     },
 
     componentWillUnmount: function () {
         $(this.props.client).off('image.containers', this.imagesChanged);
         $(this.props.client).off('pulling.containers', this.pullingChanged);
+
+        atomic.removeEventListener('vulnerableInfoChanged', this.vulnerableInfoChanged);
     },
 
-    render: function () {
-        var filtered = this.state.images.filter(function (image) {
-            return (image.RepoTags &&
-                    image.RepoTags[0].toLowerCase().indexOf(this.props.filterText.toLowerCase()) >= 0);
-        }.bind(this));
+    renderRow: function (image, withVul) {
+        var vulnerabilityColumn = '';
+
+        var vulnerableInfo = this.state.vulnerableInfos[image.Id.replace(/^sha256:/, '')];
+        if (vulnerableInfo) {
+            var count = vulnerableInfo.vulnerabilities.length;
+            if (count > 0)
+                vulnerabilityColumn = (
+                    <div>
+                        <span className="pficon pficon-warning-triangle-o"></span>
+                        &nbsp;
+                        { cockpit.ngettext('1 Vulnerability', cockpit.format('$0 Vulnerabilities', count), count) }
```

Comment:
This needs to be the other way around. The arguments to ngettext should be the strings, and ```cockpit.format``` should operate on the result. At least that's how I understand it. If this does actually work, both the scanning and translating, then let me know.